debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlBinaryProtocolFieldReader.java

```diff
@@ -74,6 +74,10 @@ protected Object readDateField(ResultSet rs, int columnIndex, Column column, Tab
         if (b == null) {
             return null; // Don't continue parsing date field if it is null
         }
+        else if (b.length() == 0) {
+            // Zero date has zero length when binary protocol uses compression.
+            return MySqlValueConverters.stringToLocalDate("0000-00-00", column, table);
```

Comment:
@vjuranek IMHO the result of this call will always be null. I suppose we should apply the same behaviour as in `io.debezium.connector.mysql.MySqlDefaultValueConverter.convertToLocalDate(Column, String)` and return Epoch time.